app/services/submit_to_app_store/nsm_payload_builder.rb

```diff
@@ -14,14 +14,22 @@ def payload
         application_id: claim.id,
         json_schema_version: 1,
         application_state: 'submitted',
-        application: data,
+        application: validated_data,
         application_risk: scorer.calculate(claim),
         application_type: 'crm7'
       }
     end
 
     private
 
+    def validated_data
+      built_data = data
```

Comment:
Why is this being copied? `validate` shouldn't mutate it, right?

Reply:
It's because `data` isn't an attribute, it's a method that does work, so we don't want to call it twice. Instead I'm assigning its output to a variable.

Reply:
Ah in that case then maybe it's cleaner to use `tap`? (untested napkin Ruby)

```ruby
construct_payload.tap do |payload|
  issues = LaaCrimeFormsCommon::Validator.validate(:nsm, payload)
  raise "Validation issues detected for #{claim.id}: #{issues.to_sentence}" if issues.any?
end
 ```

Reply:
Very good shout! Yes, have updated to use exactly that.